app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -199,6 +252,169 @@ fun HomeScreen(
+        colors =
+            TextFieldDefaults.colors(
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                disabledIndicatorColor = Color.Transparent),
+        placeholder = { Text("Search event...") },
+        trailingIcon = {
+          IconButton(
+              onClick = onSearchButtonClick, modifier = Modifier.padding(horizontal = 4.dp)) {
+                Icon(Icons.Default.Search, contentDescription = null)
+              }
+        })
+
+    // Filter button
+    Button(
+        // function setShowFilterPopUp() needs to be created in the VM
+        onClick = { setShowFilterPopUp(!showFilterPopUp) },
+        modifier = Modifier.padding(start = 8.dp)) {
+          Text("Filter")
+        }
+  }
+}
+
+@Composable
+fun FilterPopUp(
+    onRadiusChange: (Double) -> Unit,
+    onFreeSelectionChange: (Boolean) -> Unit,
+    onCategorySelectionChange: (List<EventCategory>) -> Unit,
+    modifier: Modifier = Modifier,
+    uiState: EventsOverviewUiState = EventsOverviewUiState()
+) {
+  Box(modifier = modifier) {
+    Card(
+        modifier = Modifier.padding(12.dp),
+        elevation = cardElevation(defaultElevation = 4.dp),
+    ) {
+      Column(modifier = Modifier.fillMaxSize().padding(12.dp)) {
+        // Text input for radius
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Start) {
+              Text(
+                  text = "Radius: ",
+                  style =
+                      TextStyle(
+                          fontSize = 16.sp,
+                      ))
+              TextField(
+                  value = uiState.radiusInputFilter.toString(),
+                  onValueChange = { input ->
+                    uiState.radiusInputFilter = input.toDoubleOrNull() ?: uiState.radiusInputFilter
+                  },
+                  keyboardOptions =
+                      KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number),
+                  modifier = Modifier.width(70.dp).height(10.dp),
+              )
+              Text(
+                  text = " km",
+                  modifier = Modifier.weight(1f),
+                  style =
+                      TextStyle(
+                          fontSize = 16.sp,
+                      ))
+            }
+
+        Spacer(modifier = Modifier.height(12.dp))
+
+        // Slider for free selection
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Start) {
+              Text(
+                  text = "Free:  ",
+                  style =
+                      TextStyle(
+                          fontSize = 16.sp,
+                      ))
+              Switch(
+                  checked = uiState.freeEventsFilter,
+                  onCheckedChange = { uiState.freeEventsFilter = it },
```

Comment:
And here is the reason why you should have access to a "read only" state in the view (i.e. your filter pop up in this case). You souldn't be able to modify the state directly from the view => it should be done in the viewmodel

Reply:
Resolved in PR #190